addons/stock/models/stock_move.py

```diff
@@ -275,7 +275,7 @@ def _compute_show_details_visible(self):
     @api.depends('state', 'picking_id.is_locked')
     def _compute_is_initial_demand_editable(self):
         for move in self:
-            move.is_initial_demand_editable = not move.picking_id.is_locked or move.state == 'draft'
+            move.is_initial_demand_editable = (not move.picking_id.is_locked and move.state != 'done') or move.state == 'draft'
```

Comment:
initial demand is readonly in "ready" state. unlocking the picking should allow to edit it in this state